src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -273,32 +275,40 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
 				
 				imgInfo.format = EF_R8G8B8_SRGB;
 
+				const void* planarData[] = { data , nullptr, nullptr, nullptr };
+				const size_t wholeSize = region.imageExtent.height * region.bufferRowLength;
+				const auto wholeSizeInBytes = wholeSize * getTexelOrBlockBytesize(EF_R8G8B8_SRGB);
+				uint8_t* outRGBData = _IRR_NEW_ARRAY(uint8_t, wholeSizeInBytes);
+
+				video::convertColor<EF_R8_SRGB, EF_R8G8B8_SRGB>(planarData, outRGBData, wholeSize, *reinterpret_cast<core::vector3d<uint32_t>*>(&region.imageExtent));
+
+				texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(wholeSizeInBytes);
+
+				memcpy(texelBuffer->getPointer(), outRGBData, wholeSizeInBytes);
+				_IRR_DELETE_ARRAY(outRGBData, wholeSizeInBytes); // it involves size in R8G8B8
+
+				image->setBufferAndRegions(std::move(texelBuffer), regions);
+
 				// Targa formats needs two y-axis flips. The first is a flip to get the Y conforms to OpenGL coords.
```

Comment:
it would be easier on the CPU and memory to flip before expanding to 3x the size during the convert